Source/Evolve/WarpXEvolve.cpp

```diff
@@ -466,20 +466,21 @@ WarpX::OneStep_nosub (Real cur_time)
         } else {
             amrex::Abort(Utils::TextMsg::Err("Medium for EM is unknown"));
         }
-
         FillBoundaryE(guard_cells.ng_FieldSolver, WarpX::sync_nodal_points);
+
         EvolveF(0.5_rt * dt[0], DtType::SecondHalf);
         EvolveG(0.5_rt * dt[0], DtType::SecondHalf);
         EvolveB(0.5_rt * dt[0], DtType::SecondHalf); // We now have B^{n+1}
 
         if (do_pml) {
-            FillBoundaryF(guard_cells.ng_alloc_F);
             DampPML();
             NodalSyncPML();
             FillBoundaryE(guard_cells.ng_MovingWindow);
-            FillBoundaryF(guard_cells.ng_MovingWindow);
             FillBoundaryB(guard_cells.ng_MovingWindow);
+            FillBoundaryF(guard_cells.ng_MovingWindow);
+            FillBoundaryG(guard_cells.ng_MovingWindow);
```

Comment:
Real code changes:
- removed call to `FillBoundaryF` before `DampPML`/`NodalSyncPML` (already called after those);
- added call to `FillBoundaryG` after `DampPML`/`NodalSyncPML`, like the ones for E, B, and F.

Reply:
Interesting... with which set of parameters do you think that this could have affected WarpX ?

Reply:
The missing call to `FillBoundaryG` could affect simulations wher div(B) cleaning is used in the regular domain and in the PML (G being the scalar field added to Maxwell's equations when div(B) cleaning is used, such that dG/dt = div(B)). On the other side, I think the first call to `FillBoundaryF` before the PML damping and synchronization, now removed, was simply redundant.